clue.js

```diff
@@ -0,0 +1,340 @@
+  name: 'Hall',
+  color: '#c29970',
+  image: 'assets/hall.png'
+}
+
+const lounge = {
+  name: 'Lounge',
+  color: '#d19494',
+  image: 'assets/lounge.png'
+}
+
+const diningRoom = {
+  name: 'Dining Room',
+  color: '#669966',
+  image: 'assets/diningroom.png'
+}
+
+const kitchen = {
+  name: 'Kitchen',
+  color: '#dbbdcc',
+  image: 'assets/kitchen.png'
+}
+
+const ballroom = {
+  name: 'Ballroom',
+  color: '#bdbddb',
+  image: 'assets/ballroom.png'
+}
+
+const conservatory = {
+  name: 'Conservatory',
+  color: '#92b2c9',
+  image: 'assets/conservatory.png'
+}
+
+const billiardRoom = {
+  name: 'Billiard Room',
+  color: '#667a52',
+  image: 'assets/billiardroom.png'
+}
+
+const library = {
+  name: 'Library',
+  color: '#738c80',
+  image: 'assets/library.png'
+}
+
+const study = {
+  name: 'Study',
+  color: '#7ab8b8',
+  image: 'assets/study.png'
+}
+
+const spa = {
+  name: 'Spa',
+  color: '#ccd1c7',
+  image: 'assets/spa.png'
+}
+
+const observatory = {
+  name: 'Observatory',
+  color: '#667f99',
+  image: 'assets/observatory.png'
+}
+
+const stables = {
+  name: 'Stables',
+  color: '#a6a659',
+  image: 'assets/stables.png'
+}
+
+const guestHouse = {
+  name: 'Guest House',
+  color: '#d6c2d6',
+  image: 'assets/guesthouse.png'
+}
+
+const trophyRoom = {
+  name: 'Trophy Room',
+  color: '#999966',
+  image: 'assets/trophyroom.png'
+}
+
+const carriageHouse = {
+  name: 'Carriage House',
+  color: '#b8cce0',
+  image: 'assets/carriagehouse.png '
+}
+
+// Arrays for suspects, weapons, and rooms.
+const suspects = [
+  mrGreen, 
+  profPlum, 
+  missScarlett, 
+  mrsPeacock, 
+  colMustard, 
+  mrsWhite
+]
+
+const weapons = [
+  rope, 
+  knife, 
+  candlestick, 
+  leadPipe, 
+  poison, 
+  horseshoe, 
+  hatPin, 
+  wrench, 
+  revolver
+]
+
+const rooms = [
+  hall, 
+  lounge, 
+  diningRoom, 
+  kitchen, 
+  ballroom, 
+  conservatory, 
+  billiardRoom, 
+  library, 
+  study, 
+  spa, 
+  observatory, 
+  stables, 
+  guestHouse, 
+  trophyRoom, 
+  carriageHouse
+]
+
+// THIS FUNCTION WILL RANDOMLY SELECT ONE ITEM FROM THE ARRAY THAT YOU PASS IN TO THE FUNCTION.
+const randomSelector = array => {
+  return array[Math.floor(Math.random() * array.length)];
+}
+
+// Object to hold The Mystery
+const mystery = {
+  killer: '',
+  weapon: '',
+  room: ''
+}
+
+// Funtions to select a random Killer, Weapon, and Room to add to Mystery Object and display card.
+const pickKiller = () => {
+  mystery.killer = randomSelector(suspects)
+
+  const killer = document.getElementById('killer-card')
+  const killerName = document.getElementById("killerTitle")
+  const killerPic = document.getElementById("killerImg")
+  const killerFullName = document.getElementById("killerName")
+  const killerAge = document.getElementById("killerAge")
+  const killerWrk = document.getElementById("killerOccupation")
+  const killerFavWeapon = document.getElementById("killerFavWeapon")
+  const killerDescrip = document.getElementById("killerDescription")
```

Comment:
Do these variables need to be declared inside of the function or can they be declared outside?

Reply:
I have them inside the function since I don't use the variables anywhere else. Is it better to pull them out?